pkg/controller/jobframework/reconciler.go

```diff
@@ -266,10 +266,10 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		if err := r.stopJob(ctx, job, object, wl, evCond.Message); err != nil {
 			return ctrl.Result{}, err
 		}
-		if workload.IsAdmitted(wl) {
+		if workload.HasQuotaReservation(wl) {
```

Comment:
shouldn't we unset both conditions in this case? Or is your intention that updating this one will trigger updating the Admitted condition in the next sync?

I would prefer updating both in one API call, just to be efficient in the number of API calls.